### PR TITLE
Update documentation for new version

### PR DESCRIPTION
**Pull request by ryantwolf**
## Description

Updates the documentation for the `0.3.0` release on PyPi

## Usage
N/A
## Checklist

- [x] I am familiar with the [Contributing Guide](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md).
- [x] New or Existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by ayushdg — APPROVED** [review]
lgtm! I'll hold off on merging for a bit to make merge conflicts a bit easier in #62, I'll revisit this and fix merge conflicts once those changes are merged in.

**Review by Maghoumi — APPROVED** [review]
LGTM, needs to be merged after the latest README changes go in.

**Pull request merged**
